eos-updater/poll-common.c

```diff
@@ -571,13 +573,23 @@ should_follow_checkpoint (OstreeSysroot     *sysroot,
     }
 
   /* https://phabricator.endlessm.com/T31776 */
-  if (is_eos3a_to_eos4 &&
+  if (is_conditional_upgrade_path &&
       boot_args_contain ("ro"))
     {
       *out_reason = g_strdup (_("Read-only systems are not supported in EOS 4."));
       return FALSE;
     }
 
+  g_message("booted_ref=%s, target_ref=%s", booted_ref, target_ref);
+  /* https://phabricator.endlessm.com/T32542 */
+  if (is_conditional_upgrade_path &&
+      g_str_has_suffix (booted_ref, "nexthw/eos3.9") &&
+      g_str_has_suffix (target_ref, "eos3a"))
```

Comment:
I don't think this check is necessary. We will not be publishing a redirect from nexthw/eos3.9 to eos3a.

Reply:
ok